### PR TITLE
currentRun.compiles now correctly works in toolboxes

### PR DESCRIPTION
**Pull request by xeno-by**
Another random bug uncovered and extinguished when hacking macro annots.

### CODE REVIEW

**Comment by xeno-by** [comment]
review @retronym

File: test/files/run/toolbox_current_run_compiles.scala [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+import scala.reflect.runtime.universe._
+import scala.reflect.runtime.{universe => ru}
+import scala.tools.reflect.ToolBox
+
+object Test extends App {
+  val cm = ru.runtimeMirror(getClass.getClassLoader)
+  val toolbox = cm.mkToolBox()
+  println(toolbox.eval(toolbox.parse("{ class C; pkg.Macros.compiles[C] }")))
```
Comment (retronym) [inline test/files/run/toolbox_current_run_compiles.scala]: You could include a `false` case here for completeness, e.g `compiles[Object] == false`

File: src/compiler/scala/tools/reflect/ToolBoxFactory.scala [hunk 2]
```diff
@@ -138,7 +139,9 @@ abstract class ToolBoxFactory[U <: JavaUniverse](val u: U) { factorySelf =>
         val wrapper2      = if (!withMacrosDisabled) (currentTyper.context.withMacrosEnabled[Tree] _) else (currentTyper.context.withMacrosDisabled[Tree] _)
         def wrapper       (tree: => Tree) = wrapper1(wrapper2(tree))
 
-        phase = (new Run).typerPhase // need to set a phase to something <= typerPhase, otherwise implicits in typedSelect will be disabled
+        val run = new Run
+        run.symSource(ownerClass) = NoAbstractFile
```
Comment (retronym) [inline src/compiler/scala/tools/reflect/ToolBoxFactory.scala]: Warrants a comment.

File: src/compiler/scala/tools/reflect/ToolBoxFactory.scala [hunk 3]
```diff
@@ -138,7 +139,9 @@ abstract class ToolBoxFactory[U <: JavaUniverse](val u: U) { factorySelf =>
...
+        phase = run.typerPhase // need to set a phase to something <= typerPhase, otherwise implicits in typedSelect will be disabled
```
Comment (retronym) [inline src/compiler/scala/tools/reflect/ToolBoxFactory.scala]: Is this change tested?

File: test/files/run/toolbox_current_run_compiles.scala (hunk 1 above)
Comment (xeno-by) [inline test/files/run/toolbox_current_run_compiles.scala]: Done

File: src/compiler/scala/tools/reflect/ToolBoxFactory.scala (hunk 2 above)
Comment (xeno-by) [inline src/compiler/scala/tools/reflect/ToolBoxFactory.scala]: Done

File: src/compiler/scala/tools/reflect/ToolBoxFactory.scala (hunk 3 above)
Comment (xeno-by) [inline src/compiler/scala/tools/reflect/ToolBoxFactory.scala]: This is not a change actually

**Comment by retronym** [comment]
LGTM

**Pull request merged**